common/src/main/java/io/druid/math/expr/ExprEval.java

```diff
@@ -150,7 +150,7 @@ public final double asDouble()
   {
     private DoubleExprEval(Number value)
     {
-      super(value);
+      super(Preconditions.checkNotNull(value, "value"));
```

Comment:
The value can't actually be null. For reasons of trying to make the behavior more consistent with Druid null handling in general, `null` value is always going to be a `StringExprEval`. (See that the constructors for DoubleExprEval and LongExprEval have a non-null check for value) I'll remove the null checks in toExpr, since they are pointless, given value cannot be null.